qip-0009.md

```diff
@@ -0,0 +1,529 @@
+ ```
+  QIP: 9
+  Layer: Consensus (hard fork)
+  Title: Interlinks in Quai
+  Author: gameofpointers <gameofpointers+qip@quai.org>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/quainetwork/qips/wiki/Comments:QIP-0009
+  Status: Draft
+  Type: Standards Track
+  Created: 2023-11-16
+  License: BSD-2-Clause
+ ```
+
+## Abstract
+
+This QIP defines a block interlink structure to be used in Quai as described in
+[1] with few modifications.
+
+## Motivation
+
+To Prove/Verify the Proof Of Work done on Quai since the genesis using a
+non-interactive proof of proof of work (NiPoPoW). Also verify a transaction inclusion in a Block at a depth for which the light node(SPV client) doesn't have the state to verify. This requires using only
+O(log(N)) headers instead of the whole chain history. Specifically in Quai, the plan is to use interlinks for work based
+snap sync and to use nodes as trustless RPC providers.
+
+## Specification
+
+### Definitions
+1. $Order(O)$ - Quai blockchain is merged mined hierarchy of blockchains organized into different levels, Prime, Region and Zone. Prime has a order of 0, Region has order of 1, Zone has order 2. Order of a Block(B) is defined as $O(B)$. $O(B_p)=0$, $O(B_r)=1$, and $O(B_z)=2$. The organization of the levels and merged mining is explained in the [qip0005](qip-0005.md) spec.
+
+### Overview
+
+Interlink data structure is a skip-list that helps a verifier to process a
+sparse subset of the blockchain.
+
+Proof of Work function of a Valid Block in the blockchain satisfies the Mining Target(T) i.e $id \leq T$ . 
+
+Traditional Interlinks are organized into different levels. If $id \leq
```

Comment:
> Having a bigger range between interlink levels helps with proof compression while preserving the security guarantees.

I don't think this is true. It makes the proofs larger, not smaller, for the same security guarantees.

Either way, do you have a source for this? Would like to confirm the math.

Reply:
I think that line is irrelevant now that i think about it since it depends on the k value and m value for us, so cant make any absolute statements about the size of the proof, will remove that sentence